docs/development/release-process.rst

```diff
@@ -11,10 +11,12 @@ Release Process
 
     $ nox -s release -- YY.N
 
-   You will need the password for your GPG key as well as an API token for PyPI.
+   This creates and pushes a new tag for the release
 
 #. Add a `release on GitHub <https://github.com/pypa/packaging/releases>`__.
 
+   This triggers a CI workflow which builds and publishes the package to PyPI
```

Comment:
The GitHub Environment called `pypi` must also be configured to require approvals.

Reply:
I don't think this needs to be documented in the code, but I'll update this MR's description

Reply:
Configuring it doesn't, but the presence of a human-controlled step in the release process would be good to spell out, I think.